shared/secrets_backend/tests/secrets_backend/test_base.py

```diff
@@ -0,0 +1,95 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import pytest
+
+from airflow_shared.secrets_backend.base import BaseSecretsBackend
+
+
+class _TestBackend(BaseSecretsBackend):
+    def __init__(self, conn_values: dict[str, str] | None = None, variables: dict[str, str] | None = None):
+        self.conn_values = conn_values or {}
+        self.variables = variables or {}
+
+    def get_conn_value(self, conn_id: str) -> str | None:
+        return self.conn_values.get(conn_id)
+
+    def get_variable(self, key: str) -> str | None:
+        return self.variables.get(key)
+
+
+class TestBaseSecretsBackend:
```

Comment:
I missed adding tests for connection related methods, will add in a follow up today